packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-sign-in-soc-idp/java/integrationsteps.md

```diff
@@ -1,3 +1,77 @@
 ## Integration steps
 
-The Java SDK integration steps for this use case are coming soon. See the [Okta Identity Engine Java SDK usage guide](https://github.com/okta/okta-idx-java#usage-guide) for more information.
+### Step 1: Sign in with Facebook
+
+Use [IDXAuthenticationWrapper](https://github.com/okta/okta-idx-java/blob/master/api/src/main/java/com/okta/idx/sdk/api/client/IDXAuthenticationWrapper.java) to start the sign-in process with Okta when the user goes to the sign-in page.
+
+```java
+AuthenticationResponse beginResponse = idxAuthenticationWrapper.begin()
+```
+
+Use the [AuthenticationResponse.getIdps()](https://github.com/okta/okta-idx-java/blob/master/api/src/main/java/com/okta/idx/sdk/api/response/AuthenticationResponse.java#L91) function to return a list of social Identity Provider (IdP) options configured in your org's routing rule. See [Set up your Okta org (for social Identity Providers)](/docs/guides/oie-embedded-common-org-setup/java/main/#set-up-your-okta-org-for-social-identity-providers) for IdP configuration in your org.
+
+```java
+List<Idp> idps = authenticationResponse.getIdps();
+```
+
+You need to build the sign-in HTML view page with the social sign-in options available from the list of IdPs.
+
+```javascript
+<!-- display the social login button(s), if the org is setup to support it -->
```

Comment:
Since we're detailing the Signing in with FB here, we could probably remove the other cases like Google, LinkedIn etc from the code snippet example below I feel. If other documentation samples (ASP, Golang) show all cases, then I am fine with the below for consistency sake.

Reply:
I will update this code snippet.